lib/dynamo/dynamo/jobs.py

```diff
@@ -136,3 +137,15 @@ def get_jobs_by_status_code(status_code: str, limit: int) -> List[dict]:
     )
     jobs = response['Items']
     return jobs
+
+
+def get_jobs_waiting_for_execution(limit: int) -> list[dict]:
+    table = DYNAMODB_RESOURCE.Table(environ['JOBS_TABLE_NAME'])
+    response = table.query(
+        IndexName='status_code',
+        KeyConditionExpression=Key('status_code').eq('PENDING'),
+        FilterExpression=Attr('execution_started').ne(True),
+        Limit=limit,
+    )
```

Comment:
we could call `get_jobs_by_status('PENDING')` here, and then filter by `execution_started` manually, but I'm not sure if that's an improvement or not

Reply:
I considered that, but didn't like it much. Although, I'm not sure `get_jobs_by_status_code` has any other usages now, so if there's no longer a need for it, we could just remove it.

Reply:
Agreed, I don't see any other calls to `get_jobs_by_status_code`, so I'd be in favor of eliminating that function and it's tests.